etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/UpdateReceivingFacilityWithOrderingFacilityIdentifier.java

```diff
@@ -0,0 +1,40 @@
+package gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.custom;
+
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.FhirResource;
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.RuleExecutionException;
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.CustomFhirTransformation;
+import gov.hhs.cdc.trustedintermediary.external.hapi.HapiHelper;
+import java.util.Collections;
+import java.util.Map;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Identifier;
+import org.hl7.fhir.r4.model.Organization;
+
+/**
+ * Updates the receiving facility (MSH-6) to value in Ordering Facility Name's Organization
+ * Identifier (ORC-21.10).
+ */
+public class UpdateReceivingFacilityWithOrderingFacilityIdentifier
+        implements CustomFhirTransformation {
+
+    @Override
+    public void transform(FhirResource<?> resource, Map<String, String> args)
+            throws RuleExecutionException {
+        try {
+            Bundle bundle = (Bundle) resource.getUnderlyingResource();
+            Organization receivingFacility = HapiHelper.getReceivingFacility(bundle);
+            Identifier namespaceIdentifier =
+                    HapiHelper.getHDNamespaceIdentifier(receivingFacility.getIdentifier());
+            if (namespaceIdentifier == null) {
```

Comment:
For any kind of null checks, would it be better if we use FHIRPath and the `conditions` from [validation_definitions.json](https://github.com/CDCgov/trusted-intermediary/blob/main/etor/src/main/resources/validation_definitions.json#L6) as a way to ensure the values we need to manipulate are there?